core/chains/evm/txmgr/eth_broadcaster.go

```diff
@@ -636,15 +620,12 @@ func (eb *EthBroadcaster[CHAIN_ID, HEAD, ADDR, TX_HASH, BLOCK_HASH, R, SEQ, FEE,
 		eb.SvcErrBuffer.Append(err)
 		lgr.Criticalw(`Unknown error occurred while handling eth_tx queue in ProcessUnstartedEthTxs. This chain/RPC client may not be supported. `+
 			`Urgent resolution required, Chainlink is currently operating in a degraded state and may miss transactions`, "err", err, "etx", etx, "attempt", attempt)
-		nextNonce, e := eb.ethClient.PendingNonceAt(ctx, fromAddress)
+		nextNonce, e := eb.client.PendingNonceAt(ctx, etx.FromAddress)
 		if e != nil {
 			err = multierr.Combine(e, err)
 			return errors.Wrapf(err, "failed to fetch latest pending nonce after encountering unknown RPC error while sending transaction"), true
 		}
-		if nextNonce > math.MaxInt64 {
-			return errors.Errorf("nonce overflow, got: %v", nextNonce), true
-		}
-		if int64(nextNonce) > *etx.Sequence {
+		if nextNonce > *etx.Sequence {
```

Comment:
So we still need to generalize this in future right?
I assume this is why, for now, you needed the client.PendingNonceAt function?

We may require the Sequence type to support comparison, or similar.

Reply:
correct - I think this fits into the generalizing the confirmation logic ticket: [smartcontract-it.atlassian.net/browse/BCI-860](https://smartcontract-it.atlassian.net/browse/BCI-860)

where sequence could mean anything that is ordered in some form like nonces/sequences/utxo